src/webserver.py

```diff
@@ -64,36 +75,76 @@ def command(self) -> list[str]:
             "-m",
             "gunicorn",
             "-c",
-            str(self.config_path),
+            str(self._config_path),
             self._charm_state.flask_wsgi_app_path,
         ]
 
     @property
-    def check_config_command(self) -> list[str]:
+    def _check_config_command(self) -> list[str]:
         """Returns the command to check the Gunicorn configuration.
 
         Returns:
             The command to check the Gunicorn configuration.
         """
         return self.command + ["--check-config"]
 
-    def should_reload(self, current_config: str | None) -> bool:
-        """Return whether the given current config is different from the desired configuration.
+    @property
+    def _reload_signal(self) -> signal.Signals:
+        """Get the signal used to reload the Gunicorn web server.
+
+        Returns:
+            The signal used to reload the Gunicorn web server.
+        """
+        return signal.SIGHUP
+
+    def _exec(self, command: list[str]) -> ExecResult:
+        """Execute a command inside the Flask workload container.
+
+        The command will be executed with user flask group flask inside the container.
 
         Args:
-            current_config: A string representing the current configuration.
+            command: A list of strings representing the command to be executed.
 
         Returns:
-            bool: True if the 'current_config' is different from the desired configuration and
-                a reload is need to apply the new configuration, False otherwise.
+            ExecResult: An `ExecResult` object representing the result of the command execution.
         """
-        return self.config != current_config
+        container = self._flask_container
+        exec_process = container.exec(command, user="flask", group="flask")
+        try:
```

Comment:
Why not raising this exception directly and handling it within the caller?

Reply:
I think that the command execution will be used multiple times in this class. 
So I put the exception handling inside to avoid repeating the exception handling section.